minorminer/layout/layout.py

```diff
@@ -161,19 +161,18 @@ def _graph_distance_matrix(G,
     if disconnected_distance is None:
         disconnected_distance = len(G)
 
-    return np.array(
-        [
-            [V.get(v, disconnected_distance) for v in sorted(G)]
-                 for u, V in all_pairs_shortest_path_length
-        ]
-    )
+    dist_list = [
+        (u, [V.get(v, disconnected_distance) for v in sorted(G)])
```

Comment:
The bug that you're addressing is due to sorting `G` -- node labels are not guaranteed to be comparable, so calling `sorted` here may raise an exception.  The good news is that iteration over either `G` or `all_pairs_shortest_path_length` should occur in the same order (barring a bug in `networkx`).

Reply:
Done in commit `efb48f1`.